### PR TITLE
[G2M] Typography

### LINKED ISSUE #106: [REQUEST] abstract Typography component
## Is your feature request related to a problem?

The MUI Typography adds a bottom margin with `gutterBottom` prop. However, it does not offer an option to customize the height of the margin.

---

## Describe the solution you'd like

add a more flexible way to adjust its spacing with numerical values. 

---

## Are you able to assist bring the feature to reality?

Yes 

---

### PR DESCRIPTION
**Pull request by DoParkEQ**
fixes #106

### CODE REVIEW

File: stories/typography.stories.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import Typography from '@material-ui/core/Typography'
+import Typography from '../src/typography'
+import { Divider } from '@material-ui/core'
+import theme from '../src/theme'
```
Comment (SamSverko) [inline stories/typography.stories.js:5]:
```suggestion
import { useTheme } from '@material-ui/core/styles'
```

File: stories/typography.stories.js [hunk 2]
```diff
@@ -38,24 +41,82 @@ export default {
...
+  'subtitle1',
+  'body1',
+  'button',
+  'caption',
+  'overline',
+]
 
-const Template = ({ children, variant }) => <Typography variant={variant}>{children}</Typography>
+const Template = ({ variant, children, marginBottom }) => {
+
```
Comment (SamSverko) [inline stories/typography.stories.js:76]:
```suggestion
  const theme = useTheme()
```

File: stories/typography.stories.js (hunk 1 above)
Comment (SamSverko) [reply stories/typography.stories.js:5]: By leveraging the `useTheme` hook, we don't need to re-import our theme twice (once in the global `ThemeProvider`, then another time with a manual `import theme from '...'`.

**Review by SamSverko — CHANGES_REQUESTED** [review]
This is great! Really love the "playground" section of the `Typography` story.

Since we are creating our own `Typography` component, we should export it from `./src/index.js`, so that end-users can easily import it in their projects:

`./src/index.js`

```js
import Typography from './typography'
// ...
export {
  // ...
  Typography,
}

```

**Comment by woozyking** [comment]
> This is great! Really love the "playground" section of the `Typography` story.
> 
> Since we are creating our own `Typography` component, we should export it from `./src/index.js`, so that end-users can easily import it in their projects:
> 
> `./src/index.js`
> 
> ```js
> import Typography from './typography'
> // ...
> export {
>   // ...
>   Typography,
> }
> ```

can just be `export { default as Typography } from './typography'`

**Review by SamSverko — APPROVED** [review]

**Pull request merged**